exo-go/src/logger/index.go

```diff
@@ -0,0 +1,120 @@
+package logger
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/Originate/exosphere/exo-go/src/util"
+	"github.com/fatih/color"
+	"github.com/willf/pad"
+)
+
+// Logger represents a logger
+type Logger struct {
+	Roles         []string
+	SilencedRoles []string
+	Length        int
+	Colors        map[string]color.Attribute
+}
+
+// NewLogger is Logger's constructor
+func NewLogger(roles, silencedRoles []string) *Logger {
+	logger := &Logger{Roles: roles, SilencedRoles: silencedRoles}
+	commands := []string{"exocom", "exo-run", "exo-clone", "exo-setup", "exo-test", "exo-sync", "exo-lint", "exo-deploy"}
+	logger.Colors = map[string]color.Attribute{}
+	logger.setColors(roles)
+	logger.setLength(append(roles, commands...))
+	return logger
+}
+
+func (logger *Logger) getColor(role string) (color.Attribute, bool) {
+	chosenColor, exists := logger.Colors[role]
+	return chosenColor, exists
+}
+
+func (logger *Logger) setColors(roles []string) {
+	defaultColors := []color.Attribute{color.FgMagenta, color.FgBlue, color.FgYellow, color.FgCyan}
+	for i, role := range roles {
+		logger.Colors[role] = defaultColors[i%len(defaultColors)]
+	}
+}
+
+func (logger *Logger) setLength(roles []string) {
+	for _, role := range roles {
+		if len(role) > logger.Length {
+			logger.Length = len(role)
+		}
+	}
+}
+
+func (logger *Logger) pad(text string) string {
+	return pad.Left(text, logger.Length, " ")
+}
+
+func (logger *Logger) logOutput(left, right string) {
+	if printColor, exists := logger.getColor(left); exists {
+		color.Set(printColor, color.Bold)
+		fmt.Printf("%s ", logger.pad(left))
+		color.Unset()
+		color.Set(printColor)
+		fmt.Println(right)
+		color.Unset()
```

Comment:
I agree and I did try to avoid that (see the code below) but the entire line was in bold because `color.Bold` actually affects the entire line. That's why I need line 59 to undo that effect. That was also why I said there wasn't a neat way to make the left text bold.
```	
if printColor, exists := logger.getColor(left); exists {
   boldColor := printColor.Add(color.Bold)
  fmt.Printf("%s %s\n", boldColor.Sprintf(logger.pad(left)), printColor.Sprintf(right))
}
```